src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -247,8 +245,7 @@ public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<
                 do
                 {
                     g = g._next;
-                    g.Trim();
-                    yield return resultSelector(g._key, g._elements);
+                    yield return resultSelector(g._key, g);
```

Comment:
This applies to other cases where Trim is used.
I think we should keep trimming